staff/templates/coupon/coupon_generation.html

```diff
@@ -0,0 +1,22 @@
+{% extends './base.html' %}
+{% load crispy_forms_tags %}
+{% block coupon_content %}
+<div class="py-4">
+    <h5>Generate coupons by batch</h5>
+    <form id="csv-upload-form" enctype="multipart/form-data" action="/staff/coupons/new/batch/" method="post">
+        {% csrf_token %}
+        <div class="row mt-4">
+            <label>Upload a .csv file</label>
+            {{ coupon_generation_form.csv_file }}
+            <small class="text-danger">
+                {{ coupon_generation_form.csv_file.errors|striptags }}
+            </small>
+        </div>
+        <div class="row">
+            <div class="d-flex flex-row py-5">
+            <a type="link" href="/staff/coupons/" role="button" class="btn btn-outline-secondary px-4">Cancel</a>
+            <button class="btn btn-primary mx-2 px-4" type="submit">Upload file</button>
```

Comment:
this should be "Generate coupons"?

Reply:
can, but because the label is "upload a .csv file" that's why i didn't change the button